core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -278,6 +278,18 @@ public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
     }
 
     /**
+     * Lightly connect to the specified node
+     *
+     * @param node the node to connect to
+     */
+    public void connectToNodeLight(final DiscoveryNode node) {
+        if (node.equals(localNode)) {
```

Comment:
@nik9000 has it exactly right. It was a deliberate choice to match the pattern in other methods.